crates/common/mqtt_channel/src/connection.rs

```diff
@@ -1,15 +1,20 @@
+use crate::Topic;
 use crate::{Config, ErrChannel, Message, MqttError, PubChannel, SubChannel};
 use futures::channel::mpsc;
 use futures::channel::oneshot;
 use futures::{SinkExt, StreamExt};
 use rumqttc::{
-    AsyncClient, ConnectionError, Event, EventLoop, Incoming, Outgoing, Packet, StateError,
+    AsyncClient, ClientError, ConnectionError, Event, EventLoop, Incoming, Outgoing, Packet, QoS,
+    StateError,
 };
 use std::time::Duration;
 use tokio::time::sleep;
 
 /// A connection to some MQTT server
 pub struct Connection {
+    /// mqtt_client connection
+    pub mqtt_client: AsyncClient,
+
```

Comment:
I would try hard to remove the `AsyncClient` from the `Connection` struct. One of the main design goal of this crate is to provide an abstraction to MQTT that can be tested with any actual MQTT connection. Adding this field defeats this main goal.

Reply:
Refactored the code to use mqtt for communication between the threads.  Still error handling has to be done.